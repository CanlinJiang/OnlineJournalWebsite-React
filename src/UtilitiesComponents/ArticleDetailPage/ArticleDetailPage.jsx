import React from 'react'
import {useParams} from "react-router-dom";
import {useHistory} from 'react-router-dom'
import useFetch from '../../UtilitiesFunctions/useFetch';
import postData from '../../UtilitiesFunctions/postData';
import deleteData from "../../UtilitiesFunctions/deleteData";
import styles from "./ArticleDetailPage.module.css";
import Frame from "../MainFrame/Frame"
import HTMLText from "../HTMLText/HTMLText"


export default function ArticleDetailPage() {
    const id = useParams().id;
    const history = useHistory();
   
    const {fetchedData: article, isLoading, error} = useFetch(global.url + id);

    const updateStatus = (e) => {
        article.status = e.target.value;
        const url = global.url;
        deleteData(global.url + id);
        postData({url, article});
    }

    const BodyComponent = () => (
        <div>
            {error && <div>{error}</div>}
            {isLoading && <div>Loading...</div>}
            {article && (
            <div className={styles.container}>
                <div className={styles.firstRow}>
                    <h1>{article.title}</h1>
                    <span>{article.author}</span>
                    <span>{article.date}</span>
                </div>

                <div className={styles.centerRow}><HTMLText text={article.body}/></div>

                <div className={styles.lastRow}>
                    <button className={styles.button} onClick={() => {
                        deleteData(global.url + id);
                        alert("Article deleted!");
                        history.go(-1);
                    }
                
                }>Delete</button>
                    <form className={styles.form} onChange = {updateStatus}>
                        <input type="radio" id="read" name="status" value="Read"></input><label for = "read"> Read     </label>
                        <input type="radio" id="unread" name="status" value="Unread"></input><label for = "unread"> Unread</label>      
                    </form>
                </div>
            </div>)}
        </div>
    )


    return ( 
        <Frame BodyComponent={BodyComponent}/>
    )
}
