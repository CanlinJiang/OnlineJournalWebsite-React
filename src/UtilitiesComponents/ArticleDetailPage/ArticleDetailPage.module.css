

.firstRow span{
    font-size: 1.1em;
    margin-right: 10px;
}

.firstRow h1{
    font-size: 34px;
    font-weight: bold;
    font-family: "prata",serif;
}
.firstRow {
    margin-bottom: 20px;
}

.centerRow {
    margin-top: 100px;
    font-size: 21px;
    line-height: 1.6;
    text-align: justify;
}

.centerRow p{
    margin: 30px 0px;
}

.form{
    float: right;
    padding:10px;
    margin-top: 10px;
    border: 2px solid black;
    border-radius: 5px;
}

.form input{
    margin-left: 10px;
}
.form label{
    margin-right: 10px;
}

.button {
    float: right;
    padding:12px 40px;
    margin-top: 10px;
    margin-left: 10px;
    margin-top: 9px;
    border: 2px solid black;
    border-radius: 5px;
    transition: color 1s, background-color 0.5s;
}

.button:hover{
    background-color: black;
    color:white;
}


.container {
    padding: 50px 350px 350px 350px;
}


@media (max-width: 767px) {
    .container{
        padding: 50px;
    }
}

@media (min-width: 768px) and (max-width: 1023px){
    .container{
        padding: 50px 90px 90px 90px;
    }
}

@media (min-width: 1024px) and (max-width: 1230px) {
    .container {
        padding: 50px 150px 150px 150px;
    }
}

@media (min-width: 1231px) and (max-width: 1400px) {
    .container {
        padding: 50px 250px 250px 250px;
    }
}

@media (min-width: 1501px) and (max-width: 1600px)  {
    .container {
        padding: 50px 300px 300px 300px;
    }
} 