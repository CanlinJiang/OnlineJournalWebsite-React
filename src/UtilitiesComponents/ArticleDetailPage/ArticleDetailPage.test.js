import ArticleDetailPage from "./ArticleDetailPage"
import React from "react";

describe("ArticleDetailPage tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<ArticleDetailPage />);
      expect(container).toMatchSnapshot();
    });
  
  });