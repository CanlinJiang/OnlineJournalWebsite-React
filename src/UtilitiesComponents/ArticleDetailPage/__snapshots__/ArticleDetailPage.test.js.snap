// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDetailPage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <nav
        class="navigation"
      >
        <ul
          class="leftBar"
        >
          <li>
            <a
              href="/minelist"
            >
              Mine's
            </a>
          </li>
          <li>
            <a
              href="/moodchoosing"
            >
              Moods
            </a>
          </li>
          <li>
            <a
              href="/randomwalk"
            >
              Random Walk
            </a>
          </li>
        </ul>
        <ul
          class="rightBar"
        >
          <li
            id="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            id="addNewArticle"
          >
            <a
              href="/addnewarticle"
            >
              Add New Article
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="body"
    >
      <div>
        <div>
          Loading...
        </div>
      </div>
    </div>
    <div
      class="footer"
    >
      <nav>
        <ul
          class="navigation"
        >
          <li
            id="aboutMe"
          >
            <a
              href="/aboutme"
            >
              About me
            </a>
          </li>
          <li
            id="contactMe"
          >
            <a
              href="/contactme"
            >
              Contact me
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
